pkg/cli/searcher.go

```diff
@@ -0,0 +1,215 @@
+	ctx, cancel := context.WithCancel(context.Background())
+
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+
+	go searchService.getAllImages(ctx, *servURL, username, password, *outputFormat, imageErr, &wg)
+	wg.Add(1)
+
+	var errCh chan error = make(chan error, 1)
+
+	go collectImages(outputFormat, stdWriter, &wg, imageErr, cancel, spinner, errCh)
+	wg.Wait()
+	select {
+	case err := <-errCh:
+		return true, err
+	default:
+		return true, nil
+	}
+}
+
+type imageByNameSearcher struct{}
+
+func (search imageByNameSearcher) search(params map[string]*string,
+	searchService ImageSearchService, servURL, user, outputFormat *string,
+	stdWriter io.Writer, spinner spinnerWrapper) (bool, error) {
+	if !canSearch(params, newSet("imageName")) {
+		return false, nil
+	}
+
+	username, password := getUsernameAndPassword(*user)
+	imageErr := make(chan stringPlusErr)
+	ctx, cancel := context.WithCancel(context.Background())
+
+	var wg sync.WaitGroup
+
+	wg.Add(1)
+
+	go searchService.getImageByName(ctx, *servURL, username, password, *params["imageName"], *outputFormat, imageErr, &wg)
+	wg.Add(1)
+
+	var errCh chan error = make(chan error, 1)
+	go collectImages(outputFormat, stdWriter, &wg, imageErr, cancel, spinner, errCh)
+
+	wg.Wait()
+
+	select {
+	case err := <-errCh:
+		return true, err
+	default:
+		return true, nil
+	}
+}
+
+func collectImages(outputFormat *string, stdWriter io.Writer, wg *sync.WaitGroup,
+	imageErr chan stringPlusErr, cancel context.CancelFunc, spinner spinnerWrapper, errCh chan error) {
+	var foundResult bool
+
+	defer wg.Done()
+	spinner.startSpinner()
+
+	for {
+		select {
+		case result := <-imageErr:
+			if result.Err != nil {
+				spinner.stopSpinner()
+				cancel()
+				errCh <- result.Err
+
+				return
+			}
+
+			if !foundResult && (*outputFormat == "text" || *outputFormat == "") {
+				spinner.stopSpinner()
+
+				var builder strings.Builder
+
+				printImageTableHeader(&builder)
+				fmt.Fprint(stdWriter, builder.String())
+			}
+
+			foundResult = true
+
+			fmt.Fprint(stdWriter, result.StrValue)
+		case <-time.After(waitTimeout):
+			cancel()
+			return
+		}
+	}
+}
+
+func getUsernameAndPassword(user string) (string, string) {
+	if strings.Contains(user, ":") {
+		split := strings.Split(user, ":")
+		return split[0], split[1]
+	}
+
+	return "", ""
+}
+
+type set struct {
+	m map[string]struct{}
+}
+
+func getEmptyStruct() struct{} {
+	return struct{}{}
+}
+
+func newSet(initialValues ...string) *set {
+	s := &set{}
+	s.m = make(map[string]struct{})
+
+	for _, val := range initialValues {
+		s.m[val] = getEmptyStruct()
+	}
+
+	return s
+}
+
+func (s *set) contains(value string) bool {
+	_, c := s.m[value]
+	return c
+}
+
+var (
+	ErrCannotSearch        = errors.New("cannot search with these parameters")
+	ErrInvalidOutputFormat = errors.New("invalid output format")
+)
+
+type stringPlusErr struct {
+	StrValue string
+	Err      error
+}
+
+type spinnerWrapper struct {
+	spinner          *spinner.Spinner
+	isSpinnerEnabled bool
```

Comment:
this is already a spinner, just called it "enabled"

simplify naming of spinnerWraper to spinner?